File: backend/data/projects/392a52dd-119c-46c9-9513-726e5066c289/code/frontend/src/types/tasks.ts
import { UUID } from './common';

export interface Task {
  task_id: UUID;
  family_id: UUID;
  title: string;
  description: string | null;
  assigned_to: UUID;
  due_date: string | null;
  points: number;
  status: 'pending' | 'completed' | 'in_progress';
  created_by: UUID;
  created_at: string;
  updated_at: string;
  task_category_id: UUID | null; // Link to custom task category
}

export interface TaskCreate extends Omit<Task, 'task_id' | 'created_at' | 'updated_at' | 'status' | 'assigned_to' | 'created_by' | 'family_id'> {
    assigned_to: UUID;
    created_by: UUID;
    family_id: UUID;
    task_category_id: UUID | null;
}

export interface TaskUpdate extends Partial<TaskCreate> {
    status?: 'pending' | 'completed' | 'in_progress';
}

File: backend/data/projects/392a52dd-119c-46c9-9513-726e5066c289/code/backend/src/models/tasks.py
from sqlalchemy import Column, String, ForeignKey, DateTime, UUID, Integer, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from backend.src.database import Base
import uuid


class Task(Base):
    __tablename__ = "tasks"

    task_id = Column(UUID(as_uuid=True), primary_key=True, index=True, default=uuid.uuid4)
    family_id = Column(UUID(as_uuid=True), ForeignKey("family_members.family_id"), nullable=False, index=True)
    title = Column(String, nullable=False)
    description = Column(Text, nullable=True)
    assigned_to = Column(UUID(as_uuid=True), ForeignKey("users.user_id"), nullable=False)
    due_date = Column(DATE, nullable=True)
    points = Column(Integer, nullable=False, default=0)
    status = Column(String, nullable=False, default='pending') # e.g., 'pending', 'in_progress', 'completed'
    created_by = Column(UUID(as_uuid=True), ForeignKey("users.user_id"), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    task_category_id = Column(UUID(as_uuid=True), ForeignKey('custom_categories.category_id'), nullable=True) # Link to custom categories

    # Relationships (define based on your User and Category models)
    # assigned_user = relationship("User", foreign_keys=[assigned_to])
    # creator_user = relationship("User", foreign_keys=[created_by])
    # category = relationship("CustomCategory", back_populates="tasks")

